### PR TITLE
Second part to refine nydus-image

### PR DESCRIPTION
**Pull request by jiangliu**
Second part to refine nydus-image

### CODE REVIEW

File: src/bin/nydus-image/core/context.rs [hunk 1]
```diff
@@ -37,25 +37,46 @@ use super::prefetch::{Prefetch, PrefetchPolicy};
...
 pub const BUF_WRITER_CAPACITY: usize = 2 << 17;
 
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub enum SourceType {
-    Directory,
-    StargzIndex,
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum ConversionType {
+    DirectoryToRafs,
+    DirectoryToStargz,
```
Comment (imeoer) [inline src/bin/nydus-image/core/context.rs:43]: Maybe we need to call it `EStargz`, it's a different format from stargz (it's not a blocker for this PR).

File: src/bin/nydus-image/builder/directory.rs [hunk 2]
```diff
@@ -147,39 +149,57 @@ impl Builder for DirectoryBuilder {
...
+        // Dump bootstrap file
+        let blob_table = blob_mgr.to_blob_table(ctx)?;
+        bootstrap.dump(
+            ctx,
+            &mut bootstrap_mgr.bootstrap_storage,
+            &mut bootstrap_ctx,
+            &blob_table,
+        )?;
+
         if let Some((_, blob_ctx)) = blob_mgr.get_current_blob() {
```
Comment (imeoer) [inline src/bin/nydus-image/builder/directory.rs:172]: If no blob is generated, we still need to write bootstrap data and tar header (`TAR_BOOTSTRAP_NAME`) into the blob if `inline-bootstrap` is enabled.

**Comment by imeoer** [comment]
Other LGTM!

File: src/bin/nydus-image/core/context.rs (hunk 1 above)
Comment (jiangliu) [reply src/bin/nydus-image/core/context.rs:43]: Will fix in later patch.

File: src/bin/nydus-image/builder/directory.rs (hunk 2 above)
Comment (jiangliu) [reply src/bin/nydus-image/builder/directory.rs:176]: Fixed

**Review by imeoer — APPROVED** [review]

**Pull request merged**